src/olympia/editors/models.py

```diff
@@ -758,9 +758,41 @@ def calculate_weight(self):
                         files__status=amo.STATUS_DISABLED)
                 .distinct().count() * 10, 100),
         }
+        factors.update(self.calculate_static_analysis_weight_factors())
         self.weight = sum(factors.values())
         return factors
 
+    def calculate_static_analysis_weight_factors(self):
+        """Calculate the static analysis risk factors.
+        Used by calculate_weight()."""
+        try:
+            factors = {
+                # Static analysis flags from linter:
+                # eval, evalInSandbox, document.write: 20.
+                'uses_dangerous_eval': (
+                    20 if self.check_uses_eval(self.version) else 0),
+                # Implied eval in setTimeout/setInterval/ on* attributes: 5.
+                'uses_implied_eval': (
+                    5 if self.check_uses_implied_eval(self.version) else 0),
+                # innerHTML / unsafe DOM: 20.
+                'uses_innerhtml': (
+                    20 if self.check_uses_innerhtml(self.version) else 0),
+                # custom CSP: 30.
+                'uses_custom_csp': (
+                    30 if self.check_uses_custom_csp(self.version) else 0),
+                # nativeMessaging permission: 20.
+                'uses_native_messaging': (
+                    20 if self.check_uses_native_messaging(self.version) else
+                    0),
```

Comment:
nitpick, I wonder if we can make that list static, make the separate keys `lambda` functions that accept `version` as a parameter. That might make this function easier to read, maybe. Nothing I insist on though, feel free to leave it as is.